Some nits and tweaks

gnosek opened the pull request:
<!--
Thanks for sending a pull request!

Here are some tips for you:

1. If this is your first time, please read our contributor guidelines in the [CONTRIBUTING.md](https://github.com/falcosecurity/.github/blob/master/CONTRIBUTING.md) file.
2. Please label this pull request according to what type of issue you are addressing.
3. Please add a release note in case it is needed!
4. If the PR is unfinished while opening it specify a wip in the title before the actual title, for example "wip: my awesome feature"
-->

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area api

/area client

> /area tests

> /area examples

> /area build

**What this PR does / why we need it**:

Some minor fixes after a review of the repo.

**Which issue(s) this PR fixes**:

<!--
Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
If PR is `kind/failing-tests` please post the related issues/tests in a comment and do not use `Fixes`.
-->

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below.
If the PR requires additional action from users switching to the new release, prepend the string "action required:".
For example, `action required: change the API ...`.
-->

```release-note
NONE
```

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/client-rs/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [1e360a2](https://github.com/falcosecurity/client-rs/commits/1e360a24902bfeebb1f175b3d77822ae4944fe65) Fix redundant field name in struct initialization
- [ce55960](https://github.com/falcosecurity/client-rs/commits/ce55960fa84138a1a0e49e34060838fccaf48041) Fix module name in doctests
- [5b50fe1](https://github.com/falcosecurity/client-rs/commits/5b50fe1e45f142d1c7537d9b81071216bd3f8c11) Move auth config into a separate structure
- [6b9c80a](https://github.com/falcosecurity/client-rs/commits/6b9c80af8e18850f3b59b69322702da09e6d8958) Rework error handling a bit

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **fntlnz**
You can assign the PR to them by writing `/assign @fntlnz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fclient-rs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/client-rs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["fntlnz"]} -->

poiana commented:
Welcome @gnosek! It looks like this is your first PR to falcosecurity/client-rs 🎉

gnosek committed:
Move auth config into a separate structure

This way we cannot have halfway-configured authentication.

Signed-off-by: Grzegorz Nosek <root@localdomain.pl>

gnosek committed:
Rework error handling a bit

Keep file/line number as members of the InternalError struct,
drop the whole `Error` indirection layer.

Signed-off-by: Grzegorz Nosek <root@localdomain.pl>

leodido commented on src/config.rs:
```diff
@@ -2,14 +2,20 @@ use serde_derive::{Deserialize, Serialize};
-    pub(crate) ca: Option<PathBuf>,
-    pub(crate) cert: Option<PathBuf>,
-    pub(crate) key: Option<PathBuf>,
+    #[serde(flatten)]
```
👍

leodido reviewed (dismissed):
Thanks for contributing to this! 🔥 

I really like the error refactoring btw, bravo!

Could you please just rebase it on top of the master?

/hold

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/client-rs/pull/13#pullrequestreview-368699411" title="Approved">leodido</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fclient-rs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/client-rs/blob/master/OWNERS)~~ [leodido]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
LGTM label has been added.  <details>Git tree hash: 67d7e4c73f4e2953571c31c209c0726753378d5f</details>

leodido commented:
/hold

poiana commented:
New changes are detected. LGTM label has been removed.

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **leodido**
You can assign the PR to them by writing `/assign @leodido` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fclient-rs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/client-rs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["leodido"]} -->

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **leodido**
You can assign the PR to them by writing `/assign @leodido` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fclient-rs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/client-rs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["leodido"]} -->

gnosek commented:
Done

leodido commented:
/hold cancel

On Wed, 4 Mar 2020 at 14:08, Grzegorz Nosek <notifications@github.com>
wrote:

> Done
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/falcosecurity/client-rs/pull/13?email_source=notifications&email_token=AAA5J4YOGSYCHMYKBJJDRQTRFZHFLA5CNFSM4LBCDM62YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENXYLZQ#issuecomment-594511334>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA5J446DPJNZKQR7CMDE2DRFZHFLANCNFSM4LBCDM6Q>
> .
>
-- 
L.

poiana commented:
LGTM label has been added.  <details>Git tree hash: 4e22669e6a8e84f91b7d8b221f39a62e8c006199</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/client-rs/pull/13#pullrequestreview-369429961" title="Approved">fntlnz</a>*, *<a href="https://github.com/falcosecurity/client-rs/pull/13#pullrequestreview-369429958" title="Approved">leodido</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fclient-rs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/client-rs/blob/master/OWNERS)~~ [fntlnz,leodido]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/client-rs/pull/13#pullrequestreview-369429961" title="Approved">fntlnz</a>*, *<a href="https://github.com/falcosecurity/client-rs/pull/13#pullrequestreview-369429958" title="Approved">leodido</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=falcosecurity%2Fclient-rs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/client-rs/blob/master/OWNERS)~~ [fntlnz,leodido]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
